DEVELOPING.md

```diff
@@ -0,0 +1,43 @@
+# Development guides and tips  
+
+## Linting
+
+### Run `golangci-lint`
+
+Full list linters and config can be found in `.golangci.yml`
+
+- Consider running the linter locally before commit.  
+
+  ```shell
+  make golangci-lint
+  ```
+
+- Or run the following to allow `golangci-lint` for any auto fixes (if available by linter)
+  
+  ```shell
+  golangci-lint run --fix --config .golangci.yml
+  ```
+
+### Suppress `golangci-lint` errors
+
+If you would like to suppress a false positive, consider in-line syntax.
```

Comment:
```suggestion
If you would like to suppress a false positive, consider in-line syntax. Please note that such suppressions should only be used in exceptional cases, however, rather than common practice.
```